Source/DafnyRuntime/DafnyRuntimeJava/src/main/java/dafny/DafnySequence.java

```diff
@@ -315,22 +328,35 @@ public String verbatimString(){
     }
 
     @Override
-    @SuppressWarnings("unchecked")
-    protected List<T> asList() {
-        return (List) Arrays.asList(seq);
+    Copier<T> newCopier(final int length) {
+        return new Copier<T>() {
+            private final Object[] newArray = new Object[length];
+            private int nextIndex = 0;
+
+            @Override
+            public void copyFrom(DafnySequence<T> source) {
+                if (source instanceof ArrayDafnySequence<?>) {
```

Comment:
Why isn't `ByteArrayDafnySequence` treated similarly?

Reply:
Because we can't do any better than the general case, since we'll need to box each byte individually.

Reply:
Ah, I see.  Because the underlying there is `byte[]` and we can't put those in an `Object[]`.  Thanks.